### DISCUSSION TITLE
[Components] - Snackbar - investigation

### DISCUSSION THREAD

**Discussion by Powerplex** [adevinta/spark#1872] [Investigations]
https://sonner.emilkowal.ski/

https://www.radix-ui.com/primitives/docs/components/toast#toast

https://www.magentaa11y.com/checklist-web/toast-snackbar/

https://ariakit.org/examples/dialog-react-toastify#keeping-toasts-accessible

https://fkhadra.github.io/react-toastify/introduction/

Step 1 : do a Stackblitz to try Radix toast in uncontrolled mode
Step 2 : compare https://sonner.emilkowal.ski/ and https://react-spectrum.adobe.com/react-aria/useToast.html 

We should clarify the behaviour:
- stackable or not ?
- can have actions inside or not ? (maybe only a dismiss button ? )
- positioning (corners ? centered ?) => we advise to always use the same position/corner for a given website so users knows where to look.
- timeout or not ? (it seems for now we should have a timeout) + timers will automatically pause when the user focuses or hovers over a toast.

**Comment by soykje** [comment]
Regarding specs draft [here](https://www.figma.com/file/0QchRdipAVuvVoDfTjLrgQ/Spark-Specs-and-Handoff?type=design&node-id=43405-12813&mode=design&t=EQ7VqWqYkItUReRI-0) I would have some questions to share:
- **on the positionning**, I'm not sure we should take care of elements surrounding the `SnackBar` (such as navigation, ...). We might simply deliver an API to handle vertical and horizontal positionning, based on keywords (`'top'|'bottom'`, ...), and that could also be extended with absolute values. I personnally would recommend not to provide such API, but give a default positionning (that still could be overriden through CSS classes) to avoid ending with one instance on bottom, another on top, ...
- **on stacked messages**, we should warn design team about animations in this case. Android has a queue strategy, but I'm not sure this is what we should do too...

**Comment by soykje** [comment]
After some explorations on Radix and React Spectrum libraries, I guess we can agree on going with Spectrum one :+1: Here are some pro/cons:
- Radix Toast has some technical and accessibility issues that have not been fixed for months (?)
- Radix Toast doesn't give focus back after closing last toast (a11y)
- Radix Toast doesn't implement natively any logic related to multiple toasts

vs

- Spectrum hook doesn't bring any markup but implements a lot of logic, related to multiple toasts with a queue priority concept (https://spectrum.adobe.com/page/toast/#Priority-queue)
- Spectrum is free of accessibility issues :stuck_out_tongue_closed_eyes: 

**NB:** Radix demo with stacked toasts can be seen [here](https://stackblitz.com/edit/stackblitz-starters-sfefbn?file=pages%2Findex.tsx)

**Comment by Powerplex** [comment]
Snackbar is Deprecated in favor of `Toast`. Closing.
